app/models/foreman_google/google_compute.rb

```diff
@@ -167,25 +172,28 @@ def load_image(image_id)
 
     def construct_volumes(image_id, volumes = [])
       return [Google::Cloud::Compute::V1::AttachedDisk.new(disk_size_gb: 20)] if volumes.empty?
+
       image = load_image(image_id)
 
-      # TODO: Bug, google has disk_size_gb, not size_gb
-      # TODO: Should return array of Google::Cloud::Compute::V1::AttachedDisk
-      #       when creating new host for compute resource
-      new_vol_attrs = volumes.map.with_index do |vol_attrs, i|
-        { name: "#{@name}-disk#{i + 1}",
-          size_gb: vol_attrs[:size_gb]&.to_i }
+      attached_disks = volumes.map.with_index do |vol_attrs, i|
+        name = "#{@name}-disk#{i + 1}"
+        size = (vol_attrs[:size_gb] || vol_attrs[:disk_size_gb]).to_i
+
+        Google::Cloud::Compute::V1::AttachedDisk.new(device_name: name, disk_size_gb: size)
       end
 
-      new_vol_attrs.first[:source_image] = image&.self_link
-      new_vol_attrs
+      attached_disks.first.source = image&.self_link if image&.self_link
+      attached_disks
     end
 
     # Note - GCE only supports cloud-init for Container Optimized images and
     # for custom images with cloud-init setup
-    def construct_metadata(user_data)
-      return if user_data.blank?
-      { items: [{ key: 'user-data', value: user_data }] }
+    def construct_metadata(args)
+      ssh_keys = { key: 'ssh-keys', value: "#{args[:username]}:#{args[:public_key]}" }
```

Comment:
Not sure what to think about this. When the vm is created, the ssh post-provisioning fails to connect. Yet when I access the host from gce console, the key seems to be present in root's authorized_keys file

Reply:
The instance was up, but the key it tried to use was not accepted

Reply:
CentOS7, image centos-7-v20220621, without userdata, deployed into us-central-1 as e2-micro with an ephemeral public ip.

```
D, [2022-06-22T16:43:18.306775 #261676] DEBUG -- net.ssh.authentication.methods.publickey[f8a5c]: trying publickey (d3:da:11:fc:06:6b:84:7c:e3:a8:1b:f8:f4:53:c5:8e)                                                                                                                    
2022-06-22T16:43:18 [D|app|14b6dc5e] trying publickey (d3:da:11:fc:06:6b:84:7c:e3:a8:1b:f8:f4:53:c5:8e)                                                                                                                                                                                 
D, [2022-06-22T16:43:18.306986 #261676] DEBUG -- socket[f89e4]: queueing packet nr 5 type 50 len 348                                        
2022-06-22T16:43:18 [D|app|14b6dc5e] queueing packet nr 5 type 50 len 348                                                                                                                                                                                                               
D, [2022-06-22T16:43:18.307145 #261676] DEBUG -- socket[f89e4]: sent 372 bytes                                                                                                                                                                                                          
2022-06-22T16:43:18 [D|app|14b6dc5e] sent 372 bytes                                                                                                                                                                                                                                     
D, [2022-06-22T16:43:18.485333 #261676] DEBUG -- socket[f89e4]: read 324 bytes                                                                                                                                                                                                          
2022-06-22T16:43:18 [D|app|14b6dc5e] read 324 bytes                                                                                                                                                                                                                                     
D, [2022-06-22T16:43:18.485533 #261676] DEBUG -- socket[f89e4]: received packet nr 5 type 60 len 300                                                                                                                                                                                    
2022-06-22T16:43:18 [D|app|14b6dc5e] received packet nr 5 type 60 len 300                                                                                                                                                                                                               
D, [2022-06-22T16:43:18.486967 #261676] DEBUG -- socket[f89e4]: queueing packet nr 6 type 50 len 620                                                                                                                                                                                    
2022-06-22T16:43:18 [D|app|14b6dc5e] queueing packet nr 6 type 50 len 620                                                                                                                                                                                                               
D, [2022-06-22T16:43:18.487115 #261676] DEBUG -- socket[f89e4]: sent 644 bytes                                                                                                                                                                                                          
2022-06-22T16:43:18 [D|app|14b6dc5e] sent 644 bytes                                                                                                                                                                                                                                     
D, [2022-06-22T16:43:18.618956 #261676] DEBUG -- socket[f89e4]: read 84 bytes                                                                                                                                                                                                           
2022-06-22T16:43:18 [D|app|14b6dc5e] read 84 bytes                                                                                                                                                                                                                                      
D, [2022-06-22T16:43:18.619181 #261676] DEBUG -- socket[f89e4]: received packet nr 6 type 51 len 60                                                                                                                                                                                     
2022-06-22T16:43:18 [D|app|14b6dc5e] received packet nr 6 type 51 len 60                                                                                                                                                                                                                
D, [2022-06-22T16:43:18.619336 #261676] DEBUG -- net.ssh.authentication.session[f8a20]: allowed methods: publickey,gssapi-keyex,gssapi-with-mic                                                                                                                                         
2022-06-22T16:43:18 [D|app|14b6dc5e] allowed methods: publickey,gssapi-keyex,gssapi-with-mic                                                                                                                                                                                            
D, [2022-06-22T16:43:18.619447 #261676] DEBUG -- net.ssh.authentication.methods.publickey[f8a5c]: publickey failed (d3:da:11:fc:06:6b:84:7c:e3:a8:1b:f8:f4:53:c5:8e)                                                                                                                    
2022-06-22T16:43:18 [D|app|14b6dc5e] publickey failed (d3:da:11:fc:06:6b:84:7c:e3:a8:1b:f8:f4:53:c5:8e)                                                                                                                                                                                 
E, [2022-06-22T16:43:18.619552 #261676] ERROR -- net.ssh.authentication.session[f8a20]: all authorization methods failed (tried publickey)                                                                                                                                              
2022-06-22T16:43:18 [E|app|14b6dc5e] all authorization methods failed (tried publickey)                                                                                                                                                                                                 
2022-06-22T16:43:18 [D|app|14b6dc5e] Auth failed for root at 35.202.149.249, retrying
```